Storybook: Add stories for BlockTitle Component

SainathPoojary opened the pull request:
Part of #67165  

## What?
This PR adds Storybook stories for the `BlockTitle` component to improve component documentation and testability.

## Why?
- Provide visual documentation for the BlockTitle component
- Enable interactive testing of different block title scenarios
- Improve development experience for Block Editor components

## Testing Instructions
1. Run `npm run storybook:dev`
2. Open Storybook at http://localhost:50240/
3. Verify the following stories are present and functioning:
   - Default block title rendering
   - Block title with maximum length truncation
   - Different block type titles (paragraph, heading, nested blocks)
   
## Screencast 
https://github.com/user-attachments/assets/11bb306e-4bed-4443-8e5f-9ef359352ff3

SainathPoojary committed:
Doc: Add Storybook for BlockTitle

SainathPoojary committed:
Refactor: Simplify Storybook for BlockTitle and add type summaries to controls

SainathPoojary committed:
Refactor: Updated descriptions to match the JSDoc

SainathPoojary commented:
Hi @t-hamano,

I have made the necessary changes to the PR based on the general feedback provided. Additionally, I reviewed other similar PRs with requested changes and implemented those adjustments as well. Could you kindly take a look at the PR when you have a moment?

Thank you!


https://github.com/user-attachments/assets/488f0bdc-5f75-4af2-8b7e-25a7d3514c3c

SainathPoojary committed:
Refactor: Remove comment from BlockTitle story

t-hamano commented on packages/block-editor/src/components/block-title/stories/index.story.js:
```diff
@@ -0,0 +1,108 @@
+ */
+import { registerCoreBlocks } from '@wordpress/block-library';
+import { createBlock } from '@wordpress/blocks';
+import { BlockEditorProvider } from '@wordpress/block-editor';
```
Since this component is an internal dependency, we write it as follows:

```JavaScript
/**
 * Internal dependencies
 */
import { ExperimentalBlockEditorProvider } from '../../provider';
```

t-hamano commented on packages/block-editor/src/components/block-title/stories/index.story.js:
```diff
@@ -0,0 +1,108 @@
+		),
+	],
+	argTypes: {
+		clientId: {
+			control: {
+				type: 'select',
+				labels: {
+					[ blocks[ 0 ].clientId ]: "Paragraph's Client ID",
+					[ blocks[ 1 ].clientId ]: "Heading's Client ID",
+					[ blocks[ 2 ].clientId ]: "Group's Client ID",
+				},
+			},
+			options: [
+				blocks[ 0 ].clientId,
+				blocks[ 1 ].clientId,
+				blocks[ 2 ].clientId,
+			],
+			description: 'Client ID of block.',
+			table: {
+				type: {
+					summary: 'string',
+				},
+			},
+		},
```
```suggestion
		clientId: {
			description: 'Client ID of block.',
			table: {
				type: {
					summary: 'string',
				},
			},
		},
```

Personally, I don't think this control needs to be selectable since this component is only concerned with the title string and it doesn't really matter what the block is.

t-hamano commented on packages/block-editor/src/components/block-title/stories/index.story.js:
```diff
@@ -0,0 +1,108 @@
+		},
+		maximumLength: {
+			control: {
+				type: 'number',
+				min: 5,
+				max: 50,
+				step: 1,
```
```suggestion
				type: 'number',
```

This prop has no minimum and maximum thresholds, so I think it is best to remove it.

t-hamano commented on packages/block-editor/src/components/block-title/stories/index.story.js:
```diff
@@ -0,0 +1,108 @@
+registerCoreBlocks();
+
+// Sample blocks for testing
+const blocks = [
+	createBlock( 'core/paragraph', {
+		content: 'This is a sample paragraph block.',
+		className: 'sample-paragraph',
+	} ),
+	createBlock( 'core/heading', {
+		level: 2,
+		content: 'Sample Heading Block',
+	} ),
+	createBlock( 'core/group', {
+		layout: { type: 'flex', justifyContent: 'center' },
+	} ),
+];
```
```suggestion
const blocks = [ createBlock( 'core/paragraph' ) ];
```

The attributes should have no effect on this component.

t-hamano commented on packages/block-editor/src/components/block-title/stories/index.story.js:
```diff
@@ -0,0 +1,108 @@
+
+export default meta;
+
+/**
+ * Story variations demonstrating BlockTitle component capabilities
+ */
+export const Default = {
```
```suggestion
export const Default = {
```

t-hamano reviewed (commented):
Thanks for the PR! If you have any questions about the feedback I left, feel free to ask 👍

SainathPoojary committed:
Storybook: Refactor BlockTitle story to use ExperimentalBlockEditorProvider

- Replaced BlockEditorProvider with ExperimentalBlockEditorProvider.
- Simplified blocks array to include a single paragraph block.
- Removed unnecessary client ID mappings and control options.
- Cleaned up redundant comments and controls.

SainathPoojary commented:
Hey @t-hamano,

I’ve implemented the suggested changes as recommended. Let me know if there’s anything else you’d like me to adjust. 

Thank you!

<img width="1920" alt="Screenshot 2024-12-18 at 11 19 08 AM" src="https://github.com/user-attachments/assets/f1a64aec-0aaa-4206-a2eb-92452932016c" />

t-hamano commented:
Thanks for the update!

Unfortunately, there seems to be a critical issue when the `ExperimentalBlockEditorProvider` component does not have a `settings` prop (See [this comment](https://github.com/WordPress/gutenberg/pull/66290#issuecomment-2551053441)).

Therefore, we will not be able to preview stories for this component in trunk until that issue is resolved. Sorry, but we will have to wait until the `ExperimentalBlockEditorProvider` component issue is resolved 🙇‍♂️

swissspidy commented:
You could also fix it in this PR if you want, it's a one line change. Otherwise see #68100

SainathPoojary commented:
Hey @t-hamano ,

I see that the issue might be fixed (#68100). Let me know if there’s anything else you’d like me to address or if further changes are needed!

t-hamano commented on packages/block-editor/src/components/block-title/stories/index.story.js:
```diff
@@ -0,0 +1,77 @@
+	],
+	argTypes: {
+		clientId: {
+			description: 'Client ID of block.',
```
```suggestion
			control: { type: null },
			description: 'Client ID of block.',
```

![clientid](https://github.com/user-attachments/assets/606e0244-fe4b-4f0c-a3f2-62310a82264d)

I don't think the `clientId` should be mutable, because it's generated via `createBlock()` and changing this id would cause the block title to not render.

SainathPoojary committed:
Storybook: Set clientId control to null for BlockTitle

t-hamano reviewed (approved):
LGTM 👍
